src/main/scala/BallCore/Rest/RestManager.scala

```diff
@@ -0,0 +1,73 @@
+package BallCore.Rest
+
+import BallCore.Storage.{Migration, SQLManager}
+import cats.effect.IO
+import skunk.Session
+import skunk.codec.all.*
+import skunk.implicits.*
+
+import java.util as ju
+import java.util.UUID
+
+val DEFAULT_REST = 1.0
+
+/** Manages rest related tasks and utilities.
+  */
+class RestManager()(using sql: SQLManager):
+    sql.applyMigration(
+        Migration(
+            "Initial Rest Manager",
+            List(
+                sql"""
+                    CREATE TABLE RestValues (
+                        PlayerID UUID NOT NULL,
+                        Rest DOUBLE PRECISION NOT NULL,
+                        UNIQUE(PlayerID)
+                    );
+                    """.command
+            ),
+            List(
+                sql"""DROP TABLE RestValues;""".command
+            ),
+        )
+    )
+
+    /** Stores a player's rest value in the database.
+      *
+      * @param playerId
+      *   The player's Minecraft UUID.
+      * @param rest
+      *   The player's rest value.
+      * @param session
+      *   A connection to the database.
+      */
+    def save(playerId: UUID, rest: Double)(using
```

Comment:
also i'd probably have the API to other components be more abstracted; i.e. exposes a method to "use" a rest bonus for something that returns if there was any rest to be used as a bonus

Reply:
Could you explain this one a bit more? I'm not familiar with what you're referring to.

Reply:
save/get rest would be private methods, while there would be a single public one like `def useRest(player: UUID): Option[Double]` that anyone that needed to consume rest would call